### PR TITLE
Let nested include query returns results when quoteIdentifiers is false on Postgres

### PR DESCRIPTION
**Pull request by topikachu**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does your issue contain a link to existing issue (Closes #[issue]) or a description of the issue you are solving? #6363
- [x] Have you added new tests to prevent regressions?
- [x] Have you added an entry under `Future` in the changelog?

### Description of change

The origin code change the row attributes order which makes  AbstractQuery._groupJoinData can't append the join data to the root.
Modify the code to keep the key sequence when translate the lower case key name to camelcase name

### To reviewer:
Another solution is to modify https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/query.js#L481. However, I don't want to introduce the performance overhead on other database. But the order returned by Object.keys is arbitrary. So AbstractQuery._groupJoinData is builtin on a specific JS runtime behavior and may be wrong in the future.

### CODE REVIEW

File: lib/dialects/postgres/query.js [hunk 1]
```diff
@@ -194,17 +194,17 @@ class Query extends AbstractQuery {
...
-                row[targetAttr] = row[key];
-                delete row[key];
+                return targetAttr;
+              }else{
+                return key;
               }
-            }
-          }
+            });
+          });
```
Comment (mickhansen) [inline lib/dialects/postgres/query.js]: Why change this?

File: test/integration/dialects/postgres/dao.test.js [hunk 2]
```diff
@@ -939,6 +941,144 @@ if (dialect.match(/^postgres/)) {
...
+        this.ClassStudent = this.sequelize.define('ClassStudent', {
+        }, {
+          quoteIdentifiers: false,
+	        tableName: 'class_student'
+        });
+        this.Professor.hasMany(this.Class);
+        this.Class.belongsTo(this.Professor);
+        this.Class.belongsToMany(this.Student,{through: this.ClassStudent});
+        this.Student.belongsToMany(this.Class,{through: this.ClassStudent});
+        return this.Professor.sync({ force: true })
```
Comment (sushantdhiman) [inline test/integration/dialects/postgres/dao.test.js]: Use arrow functions to minimize or eliminate `self` (where possible)

**Comment by sushantdhiman** [comment]
@topikachu Test too :)

File: lib/dialects/postgres/query.js [hunk 3]
```diff
@@ -194,17 +194,17 @@ class Query extends AbstractQuery {
...
-          for (const row of rows) {
-            for (const key of Object.keys(row)) {
+          rows = _.map(rows,(row)=> {
+            return _.mapKeys(row, (value, key)=> {
               const targetAttr = attrsMap[key];
               if (typeof targetAttr === 'string' && targetAttr !== key) {
-                row[targetAttr] = row[key];
-                delete row[key];
+                return targetAttr;
+              }else{
```
Comment (sushantdhiman) [inline lib/dialects/postgres/query.js]: I forgot about this :(, need spaces around `else`

Comment (topikachu) [inline lib/dialects/postgres/query.js]: NP. ^-^. It has conflicts now and I will refine my PR.

**Comment by felixfbecker** [comment]
@topikachu your git email is not set correctly 😉

**Pull request merged**